android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -58,6 +58,8 @@
   private static final String PWSTRIPTIME_KEY = "pwstriptime";
   private static final RegionResolver REGION_RESOLVER = new RegionResolver();
 
+  private static boolean userHasBeenWarned = false;
```

Comment:
If setPwsEndpoint returned an error code, the caller could check it and call Utils.warnUserOnMissingApiKey() or something.